app/app.go

```diff
@@ -499,42 +502,51 @@ func NewpStakeApp(
+	)
+
+	_ = app.InterchainQueryKeeper.SetCallbackHandler(liquidstakeibctypes.ModuleName, app.LiquidStakeIBCKeeper.CallbackHandler())
+
 	liquidStakeIBCModule := liquidstakeibc.NewIBCModule(app.LiquidStakeIBCKeeper)
 
 	ibcTransferHooksKeeper := ibchookerkeeper.NewKeeper()
-	app.TransferHooksKeeper = *ibcTransferHooksKeeper.SetHooks(ibchookertypes.NewMultiStakingHooks(app.LSCosmosKeeper.NewIBCTransferHooks()))
+	app.TransferHooksKeeper = *ibcTransferHooksKeeper.SetHooks(
+		ibchookertypes.NewMultiStakingHooks(
+			app.LSCosmosKeeper.NewIBCTransferHooks(),
+			app.LiquidStakeIBCKeeper.NewIBCTransferHooks(),
+		),
+	)
 
-	var transferStack porttypes.IBCModule
-	transferStack = transfer.NewIBCModule(app.TransferKeeper)
+	var transferStack porttypes.IBCModule = transfer.NewIBCModule(app.TransferKeeper)
 	transferStack = ibchooker.NewAppModule(app.TransferHooksKeeper, transferStack)
 	transferStack = ibcfee.NewIBCMiddleware(transferStack, app.IBCFeeKeeper)
 
-	var icaHostStack porttypes.IBCModule
-	icaHostStack = icahost.NewIBCModule(app.ICAHostKeeper)
+	var icaHostStack porttypes.IBCModule = icahost.NewIBCModule(app.ICAHostKeeper)
 	icaHostStack = ibcfee.NewIBCMiddleware(icaHostStack, app.IBCFeeKeeper)
 
-	// Information will flow: ibc-port -> icaController -> lscosmos. -> LiquidStakeIBCKeeper.
-	//lscosmosModule := lscosmos.NewAppModule(appCodec, app.LSCosmosKeeper, app.AccountKeeper, app.BankKeeper)
-	//icaControllerIBCModule := icacontroller.NewIBCModule(lscosmosModule, app.ICAControllerKeeper)
-
-	var icaControllerStack porttypes.IBCModule
-	icaControllerStack = liquidStakeIBCModule
-	//TODO evaluate if lscosmos can be dropped after migration
-	icaControllerStack = lscosmos.NewAppModule(appCodec, icaControllerStack, app.LSCosmosKeeper, app.AccountKeeper, app.BankKeeper)
+	var icaControllerStack porttypes.IBCModule = liquidStakeIBCModule
+	//icaControllerStack = lscosmos.NewAppModule(appCodec, icaControllerStack, app.LSCosmosKeeper, app.AccountKeeper, app.BankKeeper)
 	icaControllerStack = icacontroller.NewIBCMiddleware(icaControllerStack, app.ICAControllerKeeper)
-	//icaControllerStack = ibcfee.NewIBCModule(icaControllerStack, app.IBCFeeKeeper)
 
-	// This module is not being used for any routing, can be removed, only part of ModuleManager.
-	// using ibcTransferHooksMiddleware instead.
-	//routerModule := router.NewAppModule(app.RouterKeeper, transferIBCModule)
-	// create static IBC router, add transfer route, then set and seal it
 	ibcRouter := porttypes.NewRouter()
-	ibcRouter.AddRoute(icahosttypes.SubModuleName, icaHostStack).
+	ibcRouter.
 		AddRoute(ibctransfertypes.ModuleName, transferStack).
+		AddRoute(icahosttypes.SubModuleName, icaHostStack).
 		AddRoute(icacontrollertypes.SubModuleName, icaControllerStack).
-		AddRoute(lscosmostypes.ModuleName, icaControllerStack)
+		AddRoute(liquidstakeibctypes.ModuleName, icaControllerStack)
+
+	//AddRoute(lscosmostypes.ModuleName, icaControllerStack)
```

Comment:
@puneet2019 do you want to add this back since we are merging into main ?

Reply:
nope, let it be.. liquidstake ibc might get some problems because of it.. may be just add a // TODO: Migrate